gateway/system/system.go

```diff
@@ -154,6 +154,7 @@ func NewSystem(opts *SystemOptions) (*System, error) {
 	// data api
 	sh := dataapiv1.NewStrictHandlerWithOptions(dapiImpl.DataApiV1Server, []nethttp.StrictHTTPMiddlewareFunc{
 		dapiimpl.NewErrorHandler(opts.Logger),
+		dapiimpl.NewTlsConnStateHandler(),
```

Comment:
Should this only be installed if client certificate authentication is enabled?  I'm not sure what the correct handling is meant to be where client certificate auth is not intended to be used, but a client sends one.